draft-ietf-tsvwg-sctp-dtls-chunk.md

```diff
@@ -1217,31 +1217,438 @@ some extension of the socket API.
+
+Using this socket option allows to remove a particular set of keys used for
+receiving DTLS chunks.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_dtls_keys_id {
+	sctp_assoc_t sdki_assoc_id;
+	uint8_t sdki_restart;
+	uint8_t sdki_conn_id_len;
+	uint16_t sdki_unused;
+	uint64_t sdki_epoch;
+	uint8_t *sdki_conn_id;
+}
+~~~
+
+``sdki_assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``sdki_restart``:
+: If the value is ``0``, the regular keys are removed, if a value different
+  from ``0`` is used, the restart keys are removed.
+
+``sdki_conn_id_len``:
+: The length of the DTLS connection identifier specified in ``sdki_conn_id``.
+
+``sdki_unused``:
+: This field is ignored.
+
+``sdki_epoch``:
+: The epoch for which the keys are removed.
+
+``sdki_conn_id``:
+: A pointer to the connection identifier for which the keys are removed.
+  Using ``NULL`` means that no connection identifier is used.
+
+This socket option can only be used on SCTP endpoints in states other then
+``SCTP_CLOSED``, ``SCTP_LISTEN``, ``SCTP_COOKIE_WAIT`` and
+``SCTP_COOKIE_ECHOED``.
+
+### Set or Get Protection Enforcement (``SCTP_DTLS_ENFORCE_PROTECTION``)
+
+Enabling this socket option on an SCTP endpoint enforces that only
+protected SCTP packets are processed anymore. All received packets with the
+first chunk not being an INIT chunk, INIT ACK chunk, or DTLS chunk will be
+silently discarded.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_assoc_value {
+	sctp_assoc_t assoc_id;
+	uint32_t assoc_value;
+};
+~~~
+
+``assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``assoc_value``:
+  The value `0` represents that the option is off, any other value represents
+  that the option is on.
+
+This socket option is off by default on any SCTP endpoint.
+Once protection has been enforced by enabling this socket option on an
+SCTP endpoint, it cannot be disabled again.
+Whether protection has been enforced on an SCTP endpoint can be queried
+in any state other than ``SCTP_CLOSED``.
+Protection can be enforced in any state other than ``SCTP_CLOSED``,
+``SCTP_COOKIE_WAIT`` and ``SCTP_COOKIE_ECHOED``.
+
+### Get Statistic Counters (``SCTP_DTLS_STATS``)
+
+This socket options allows to get various statistic counters for a
+specific SCTP endpoint.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_dtls_stats {
+	sctp_assoc_t sds_assoc_id;
+	uint32_t sds_dropped_unprotected;
+	uint32_t sds_aead_failures;
+	/* There will be added more fields before the WGLC. */
```

Comment:
So AEAD failures are for received packets. But, the abstract API defines "Get Q" which is counting number of invocations for sent packets. So these are directional.

Reply:
So I did add an issue for this question #80 so we will not forget it.